### DISCUSSION TITLE
Compressed archive handler disappeared: no right-click + "extract here" anymore

### DISCUSSION THREAD

**Discussion by stradicat** [elementary/files#2020] [Q&A] [answered]
### What Happened?

I noticed today, after the major updates from a couple of weeks ago, that I can't right-click + "extract here" on zip or tgz files anymore.
The handler is gone.

### Steps to Reproduce

1. Create or download a zip file
2. Open Files
3. Right-click to pick "extract here"
4. See that the option isn't there anymore

### Expected Behavior

Right-clicking opens a context menu with the "extract here" option

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

**Comment by jeremypw** [comment]
The contract that provided that function has been deprecated as it is not compatible with Flatpak.  It will be replaced by a portal.  Elementary is working on its own version of FileRoller (https://github.com/elementary/fileroller)

**Comment by Marukesu** [comment]
sorry for the miss click.

> The contract that provided that function has been deprecated as it is not compatible with Flatpak. It will be replaced by a portal. Elementary is working on its own version of FileRoller (https://github.com/elementary/fileroller)

it should be provided with the flatpak deb package no?

**Comment by danirabbit** [comment]
@Marukesu the contracts are shipped with the deb package for fileroller now: https://code.launchpad.net/~elementary-os/+archive/ubuntu/stable/+build/22355838

This should be pulled in via metapackages, so maybe this is a faulty upgrade?

**Comment by ibrahimrahman** [comment]
Happened to me as well. I ended up reinstalling file-roller

**Comment by stradicat** [comment]
@ibrahimrahman question: did you reinstall `file-roller`from flatpak/appcenter, or `apt`?

**Comment by ibrahimrahman** [comment, marked as answer]
@dennismayr apt. Don't install `file-roller`. Install `org.gnome.fileroller` instead.

`sudo apt install org.gnome.fileroller`

**Comment by stradicat** [comment]
@ibrahimrahman that was it, thanks for the tip!

**Comment by ghost** [comment]
Seems to work now, installed elementary yesterday.
Downloaded a zip, right clicked it to find the extract here at the bottom of the context menu under compress.
